### DISCUSSION TITLE
Fill a drawing with solid color

### DISCUSSION THREAD

**Discussion by nriesco** [ImageMagick/ImageMagick#4099] [Help]
Hi, I'm new to ImageMagick and I was wondering how can I fill a drawing with a solid color. The drawing has a transparent background, and could be any shape (assume is "closed").

The input would be mask1.png and the desired output mask1result.png (just one color, think that the blue portion is really red).

I can also assume I start filling from the center of the image, so the problem is take this point, fill it until you reach "the borders".

Any ideas on how to solve this?
Thanks!

Input (white => is transparent)
![mask1](https://user-images.githubusercontent.com/1995514/130325306-7e4225ad-5917-400c-adf4-b462f55157b1.png)

Output (assume the blue portions are red, also white is really transparent)
<img width="670" alt="mask1result" src="https://user-images.githubusercontent.com/1995514/130325321-9a44ee9d-b1de-46f8-a5a6-b0debcc51ce8.png">

**Comment by fmw42** [comment]
You can do that by flood filling at the center of the image with blue as follows. I assume a Unix platform with Imagemagick 7.

![img](https://user-images.githubusercontent.com/13458218/130326135-f4272158-f849-47fd-8e75-49241f8d6e1b.png)


`magick img.png -channel rgba -fill "rgba(0,0,255,1)" -draw "color %[fx:w/2],%[fx:h/2] floodfill" result.png`


![result](https://user-images.githubusercontent.com/13458218/130326128-d5069812-719e-429d-8627-a2230f3d07df.png)

**Reply by nriesco** [reply]
Thanks a lot. That should work!!!

I tried to run it my computer (OSX) but I get an error `convert: non-conforming drawing primitive definition 'color' @ error/draw.c/RenderMVGContent/4471.` I guess I should check my brew installation.

The version I'm using is: `Version: ImageMagick 7.1.0-4 Q16 x86_64 2021-07-18 https://imagemagick.org`

**Reply by fmw42** [reply]
That should work fine. I am on OSX Sierra with IM 7.1.0.4 Q16. Did you type it exactly?  Did you rename the file? Do you have double quotes that are straight quotes?

**Reply by nriesco** [reply]
My bad, I was using `convert` instead of `magick`, now it works perfectly fine!!! 
Thanks again!
